aws/resource_aws_acm_certificate_validation.go

```diff
@@ -0,0 +1,150 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/acm"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsAcmCertificateValidation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAcmCertificateValidationCreate,
+		Read:   resourceAwsAcmCertificateValidationRead,
+		Delete: resourceAwsAcmCertificateValidationDelete,
+
+		Schema: map[string]*schema.Schema{
+			"certificate_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"validation_record_fqdns": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"timeout": {
```

Comment:
Generally speaking, we should probably use the built-in schema timeout functionality instead of our own attribute for consistency with other resources and handling validation.

```
&schema.Resource{
  # ... other config ...
  Timeouts: &schema.ResourceTimeout{
    Create: schema.DefaultTimeout(45 * time.Minute),
  }
}
# usage
resource.Retry(d.Timeout(schema.TimeoutCreate), func() ...)
```

Reply:
Done.